compiler/src/dotty/tools/dotc/parsing/Parsers.scala

```diff
@@ -2248,15 +2248,20 @@ object Parsers {
           newExpr()
         case MACRO =>
           val start = in.skipToken()
-          val call = ident()
-          // The standard library uses "macro ???" to denote "fast track" macros
-          // hardcoded in the compiler, don't issue an error for those macros
-          // since we want to be able to compile the standard library.
-          if (call `ne` nme.???)
-            syntaxError(
-              "Scala 2 macros are not supported, see https://dotty.epfl.ch/docs/reference/dropped-features/macros.html",
-              start)
-          unimplementedExpr
+          val call = simpleExpr()
+          call match
+            case Ident(name) if name eq nme.??? =>
+              // The standard library uses "macro ???" to denote "fast track" macros
+              // hardcoded in the compiler, don't issue an error for those macros
+              // since we want to be able to compile the standard library.
+              MacroTree(call)
+            case _ if ctx.settings.YScala2ExperimentalCompat.value =>
```

Comment:
Not even sure we need that flag. Why not just always enable it? If we do have the flag, then we should have an error message here stating that this construct is only supported under it.

Reply:
This is a temporary measure until we check that
* A Scala 2 macro has an equivalent Scala 2 inline method
* The signature of the macro body is correct

We will be able to remove the flag once these checks are performed.

Reply:
Now that I added the first check it should be safe to allow it without the flag. I removed the flag.

Reply:
Without the flag, I had to disable some tests form semanticdb.